src/utils/API.js

```diff
@@ -0,0 +1,25 @@
+import axios from 'axios';
+
+const API = axios.create({
+    baseURL: 'http://localhost:3000/api/',
+});
+
+const publicRoutes = [
+    '/auth/login',
+    '/auth/register',
+]
+
+API.interceptors.request.use(
```

Comment:
better option was to include authorization at config header of axios. interceptors can be slow sometimes as before making request to server it has to pass through the function. but in case of config header you have set the axios instance once.